api/observability/v1/output_types.go

```diff
@@ -0,0 +1,714 @@
+	Compression string `json:"compression,omitempty"`
+}
+
+type Elasticsearch struct {
+	URLSpec `json:",inline"`
+
+	// Authentication sets credentials for authenticating the requests.
+	//
+	// +kubebuilder:validation:Optional
+	Authentication *HTTPAuthentication `json:"authentication,omitempty"`
+
+	// Tuning specs tuning for the output
+	//
+	// +kubebuilder:validation:Optional
+	Tuning *ElasticsearchTuningSpec `json:"tuning,omitempty"`
+
+	// defaults to: log_type-write
+	IndexSpec `json:",inline"`
+
+	// Version specifies the version of Elasticsearch to be used.
+	// Must be one of: 6-8, where 8 is the default
+	//
+	// +kubebuilder:validation:Minimum:=6
+	// +kubebuilder:validation:Maximum:=8
+	// +kubebuilder:default:=8
+	Version int `json:"version,omitempty"`
+}
+
+// GoogleCloudLoggingAuthentication contains configuration for authenticating requests to a GoogleCloudLogging output.
+type GoogleCloudLoggingAuthentication struct {
+	// Credentials points to the secret containing the `google-application-credentials.json`.
+	//
+	// +kubebuilder:validation:Required
+	Credentials *SecretKey `json:"credentials"`
+}
+
+// GoogleCloudLogging provides configuration for sending logs to Google Cloud Logging.
+// Exactly one of billingAccountID, organizationID, folderID, or projectID must be set.
+type GoogleCloudLogging struct {
+	// Authentication sets credentials for authenticating the requests.
+	//
+	// +kubebuilder:validation:Optional
+	Authentication *GoogleCloudLoggingAuthentication `json:"authentication,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	BillingAccountID string `json:"billingAccountId,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	OrganizationID string `json:"organizationId,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	FolderID string `json:"folderId,omitempty"`
+
+	// +kubebuilder:validation:Optional
+	ProjectID string `json:"projectId,omitempty"`
+
+	// LogID is the log ID to which to publish logs. This identifies log stream.
+	LogID string `json:"logId,omitempty"`
+}
+
+type HttpTuningSpec struct {
+	BaseOutputTuningSpec `json:",inline"`
+
+	// Compression causes data to be compressed before sending over the network.
+	//
+	// +kubebuilder:validation:Enum:=none;gzip;snappy;zlib
+	// +kubebuilder:default:=none
+	Compression string `json:"compression,omitempty"`
+}
+
+// HTTP provided configuration for sending json encoded logs to a generic HTTP endpoint.
+type HTTP struct {
+	URLSpec `json:",inline"`
+
+	// Authentication sets credentials for authenticating the requests.
+	//
+	// +kubebuilder:validation:Optional
+	Authentication *HTTPAuthentication `json:"authentication,omitempty"`
+
+	// Tuning specs tuning for the output
+	//
+	// +kubebuilder:validation:Optional
+	// +nullable
+	Tuning *HttpTuningSpec `json:"tuning,omitempty"`
+
+	// Headers specify optional headers to be sent with the request
+	//
+	// +kubebuilder:validation:Optional
+	Headers map[string]string `json:"headers,omitempty"`
```

Comment:
Generally it's best to avoid maps in CRDs, they make it hard to make structural schemas. Create instead a list with a struct type with a key and value
```
headers:
- name: ...
   values: []
```